table-of-contents.php

```
@@ -10,130 +10,7 @@
-					$toc .= '<li><a href="#' . sanitize_title_with_dashes( $item[2] ) . '">' . $item[2] . '</a></li>';
-				}
-				$toc .= '</ul>';
-				$toc .= '</div>';
-			}
-
-			return $toc . $content;
-		}
-
-		/**
-		 * Add an overview to the top of the page.
-		 *
-		 * @param string $content Page content.
-		 *
-		 * @return string
-		 */
-		public static function add_overview_h3( $content ) {
-			$h3s = self::get_tags( 'h3', $content );
-			if ( ! empty( $h3s ) ) {
-				$content = "<h3>Overview</h3>\n" . $content;
-			}
-			return $content;
-		}
-
-		/**
-		 * Add IDs and anchor links to the headings.
-		 *
-		 * @param string $tag     Tag to add anchor to.
-		 * @param string $content Post content.
-		 *
-		 * @return string
-		 */
-		private static function add_ids_and_jumpto_links( $tag, $content ) {
-			$items = self::get_tags( $tag, $content );
-			$first = true;
-
-			$matches = array();
-			$replacements = array();
-
-			foreach ( $items as $item ) {
-				$replacement = '';
-				$matches[]   = $item[0];
-				$id          = sanitize_title_with_dashes( $item[2] );
-
-				if ( ! $first ) {
-					$replacement .= '<p class="toc-jump"><a href="#content">&uarr; Top &uarr;</a></p>';
-				} else {
-					$first = false;
-				}
-
-				$replacement   .= sprintf( '<%1$s id="%2$s">%3$s <a href="#%2$s" class="anchor">#</a></%1$s>', $tag, $id, $item[2] );
-				$replacements[] = $replacement;
-			}
-
-			$content = str_replace( $matches, $replacements, $content );
-
-			return $content;
-		}
-
-		/**
-		 * Get the tags from post_content
-		 *
-		 * @param string $tag     Tag to search for.
-		 * @param string $content Post content.
-		 *
-		 * @return array
-		 */
-		private static function get_tags( $tag, $content = '' ) {
-			if ( empty( $content ) ) {
-				$content = get_the_content();
-			}
-			preg_match_all( "/(<{$tag}>)(.*)(<\/{$tag}>)/", $content, $matches, PREG_SET_ORDER );
-			return $matches;
-		}
-	}
-
+if ( ! class_exists( 'Table_Of_Contents' ) ) {
```

I've always wondered about the actual benefit of this - as in, if there is another class with exactly the same name, surely it would be better to give a fatal error (especially now WP core handles fatals on plugin activations better now), rather than silently fail and potentially then wondering why the plugin activated, but was, in this case, never initialised?

---

That's a good point. Can run a test to see what happens.